demo/fake_server.ex

```diff
@@ -56,22 +56,97 @@ defmodule Wanda.Executions.FakeServer do
 
   defp complete_fake_execution(execution_id, group_id, targets) do
     result = Enum.random([:passing, :warning, :critical])
-    check_results = build(:check_results_from_targets, targets: targets, result: result)
+    check_results = check_results_from_targets(targets, result)
 
     build_result =
       build(:result,
         check_results: check_results,
         execution_id: execution_id,
-        group_id: group_id,
-        result: result
+        group_id: group_id
       )
 
     Executions.complete_execution!(
       execution_id,
-      build_result
+      Map.put(build_result, :result, get_result(build_result))
     )
 
     execution_completed = Messaging.Mapper.to_execution_completed(build_result)
     :ok = Messaging.publish("results", execution_completed)
   end
+
+  defp get_result(%Result{check_results: check_results}) do
+    check_results
+    |> Enum.map(& &1.result)
+    |> Enum.map(&{&1, result_weight(&1)})
+    |> Enum.max_by(fn {_, weight} -> weight end)
+    |> elem(0)
+  end
+
+  defp result_weight(:critical), do: 2
+  defp result_weight(:warning), do: 1
+  defp result_weight(:passing), do: 0
+
+  defp check_results_from_targets(targets, result) do
+    targets
+    |> Enum.flat_map(& &1.checks)
+    |> Enum.uniq()
+    |> Enum.map(fn check_id ->
+      check_targets = Enum.filter(targets, &(check_id in &1.checks))
+      check_result_from_target(check_id, check_targets, result)
+    end)
+  end
+
+  defp expectation_evaluations_from_check(check, result) do
+    Enum.map(check.expectations, fn expectation ->
+      case expectation.type do
+        :expect ->
+          build(:expectation_evaluation,
+            type: :expect,
+            name: expectation.name,
+            return_value: result == :passing
+          )
+
+        :expect_same ->
+          build(:expectation_evaluation,
+            type: :expect_same,
+            name: expectation.name,
+            return_value:
+              if result == :passing do
+                "same_value"
+              else
+                Faker.StarWars.quote()
+              end
+          )
+      end
+    end)
+  end
+
+  defp check_result_from_target(check_id, check_targets, result) do
+    {_, check} = Catalog.get_check(check_id)
+    expectation_evaluations = expectation_evaluations_from_check(check, result)
+
+    expectation_results =
+      Enum.map(
+        expectation_evaluations,
+        &build(:expectation_result, name: &1.name, type: &1.type, result: result == :passing)
+      )
+
+    build(:check_result,
+      check_id: check_id,
+      agents_check_results:
+        Enum.map(check_targets, fn target ->
+          build(:agent_check_result,
+            agent_id: target.agent_id,
+            facts:
+              Enum.map(check.facts, fn fact ->
+                build(:fact, check_id: check_id, name: fact.name)
+              end),
+            expectation_evaluations: expectation_evaluations,
+            values: check.values
```

Comment:
How is this `check.values` working?
Is this just sending the same `values` field from the check itself?
We don't have to do this. The `values` are transformed during the evaluation. 
We should be sending something like:
```
values: [
{
name: faked_name,
value: some_value_from_the_check_values (it could be the default one)
}
]
```